File: backend/app/schemas/mock_interview.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class StartRequest(BaseModel):
    resume_id: int
    job_id: int
    total_questions: int = 5


class StartResponse(BaseModel):
    session_id: int
    question: str
    turn: int
    total_questions: int


class RespondRequest(BaseModel):
    answer: str


class AnswerFeedback(BaseModel):
    score: int               # 1-10
    feedback: str
    strengths: list[str]
    improvements: list[str]


class FinalReport(BaseModel):
    overall_score: float
    strengths: list[str]
    improvements: list[str]
    recommendation: str      # "Strong Hire" | "Hire" | "No Hire"


class RespondResponse(BaseModel):
    turn: int
    question: Optional[str]
    feedback: AnswerFeedback
    is_complete: bool
    final_report: Optional[FinalReport]


class MessageOut(BaseModel):
    turn: int
    question: str
    user_answer: Optional[str]
    ai_feedback: Optional[str]
    score: Optional[int]
    created_at: datetime

    class Config:
        from_attributes = True


class SessionOut(BaseModel):
    id: int
    resume_id: int
    job_id: int
    status: str
    total_questions: int
    created_at: datetime
    messages: list[MessageOut]

    class Config:
        from_attributes = True


class SessionSummary(BaseModel):
    id: int
    status: str
    total_questions: int
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/mockInterview.ts
import axios from 'axios';
import { useAuthStore } from '../store/authStore';

const BASE = import.meta.env.VITE_API_URL || "http://localhost:8000";;

function authHeaders() {
  const token = useAuthStore.getState().token;
  return { Authorization: `Bearer ${token}` };
}

export interface StartRequest {
  resume_id: number;
  job_id: number;
  total_questions: number;
}

export interface StartResponse {
  session_id: number;
  question: string;
  turn: number;
  total_questions: number;
}

export interface AnswerFeedback {
  score: number;
  feedback: string;
  strengths: string[];
  improvements: string[];
}

export interface FinalReport {
  overall_score: number;
  strengths: string[];
  improvements: string[];
  recommendation: 'Strong Hire' | 'Hire' | 'No Hire';
}

export interface RespondResponse {
  turn: number;
  question: string | null;
  feedback: AnswerFeedback;
  is_complete: boolean;
  final_report: FinalReport | null;
}

export interface SessionSummary {
  id: number;
  status: string;
  total_questions: number;
  created_at: string;
}

export interface MessageOut {
  turn: number;
  question: string;
  user_answer: string | null;
  ai_feedback: string | null;
  score: number | null;
  created_at: string;
}

export interface SessionOut {
  id: number;
  resume_id: number;
  job_id: number;
  status: string;
  total_questions: number;
  created_at: string;
  messages: MessageOut[];
}

export const startSession = async (data: StartRequest): Promise<StartResponse> => {
  const res = await axios.post(`${BASE}/mock-interview/start`, data, { headers: authHeaders() });
  return res.data;
};

export const respondToQuestion = async (
  sessionId: number,
  answer: string
): Promise<RespondResponse> => {
  const res = await axios.post(
    `${BASE}/mock-interview/${sessionId}/respond`,
    { answer },
    { headers: authHeaders() }
  );
  return res.data;
};

export const listSessions = async (): Promise<SessionSummary[]> => {
  const res = await axios.get(`${BASE}/mock-interview/`, { headers: authHeaders() });
  return res.data;
};

export const getSession = async (sessionId: number): Promise<SessionOut> => {
  const res = await axios.get(`${BASE}/mock-interview/${sessionId}`, { headers: authHeaders() });
  return res.data;
};

export const deleteSession = async (sessionId: number): Promise<void> => {
  await axios.delete(`${BASE}/mock-interview/${sessionId}`, { headers: authHeaders() });
};